### ISSUE TITLE
False negative `used-before-assignment` when variable is only defined in `except` clause

### ISSUE THREAD

**Issue by gcolombeau** [PyCQA/pylint#4761]
### Bug description

```python
"""used-before-assignment (E0601)"""

def undefined_variable():
    """used-before-assignment (E0601)"""
    try:
        pass
    except ValueError:
        some_message = 'some message'

    if not some_message:
        return 1

    return some_message
```

### Command used

```shell
pylint e0601.py
```

### Pylint output

```shell
------------------------------------
Your code has been rated at 10.00/10
```

### Expected behavior

Pylint should report `used-before-assignment` on lines 10 and 13.

### Pylint version

```shell
pylint 2.9.5
astroid 2.6.5
Python 3.6.10 (default, Jan 31 2020, 10:49:38) 
[GCC 7.4.0]
```
